### DISCUSSION TITLE
Xdrip Plus Wakelocks

### DISCUSSION THREAD

**Discussion by nekromantik** [NightscoutFoundation/xDrip#2631] [Q&A]
Hello so I got replacement phone a pixel 7 pro and installed latest version of xdrip plus. I use dexcom g7 so have to use companion mode on xdrip however I see 3 Wakelocks from xdrip causing battery drain. One of them is g5 wakelock. Is there a way to get rid of these? Thanks

**Comment by nekromantik** [comment]
Anyone?

**Reply by Navid200** [reply]
Can you explain how you have established battery drain?
Are you including the battery drain of the phone screen and everything else or only the apps.

Often individuals only look at the list of the apps and see that xDrip consumes more than every other app and conclude that xDrip consumes too much.
Only after they include all the system elements on the list, do they realize that it makes no differences because it's their phone screen that effectively consumes all the battery and changing how much battery xDrip needs is not going to change anything.

**Reply by nekromantik** [reply]
I wasn't looking at android battery use screen.
I use better battery stats to check Wakelocks and xdrip has Wakelocks near top of the list

**Reply by jamorham** [reply]
Can you provide the exact wakelock name concerned? xDrip stacks wakelocks so many different ones can overlap which most battery apps don't understand how to correctly deduce the time spent. Wakelocks have previously been extensively audited for best battery performance but I am happy to investigate any particular wakelock which might be consuming more than it should.

**Reply by nekromantik** [reply]
So main two are g5 insert which had most battery use and after that it's NotifySyncBgr-0

**Reply by jamorham** [reply]
Those two wakelocks definitely stack and typically it should be approx 4 seconds per reading, so about 20 minutes per 24 hour period. Are you seeing more than that for any of those individually?

**Reply by nekromantik** [reply]
![Screenshot_20230120-155815](https://user-images.githubusercontent.com/90213527/213745312-6d5df92d-ec8c-4696-9825-061ba6fb3d73.png)

**Reply by nekromantik** [reply]
That's right after a charge

**Reply by jamorham** [reply]
I think that is showing 3 seconds in a 5 minute period so that would be 14 minutes of awake time in a 24 hour period. But really this data is only meaningful over a much longer period  - so see if you can gather that. xDrip will show as an always active app because it is always active but that doesn't mean it is consuming battery or waking the device when it is sleeping which would require wakelocks.

**Reply by nekromantik** [reply]
Is the g5 wakelock needed if you don't have a dexcom g5? I would assume if your not using xdrip as collector then battery use will be minimal.

**Reply by Navid200** [reply]
It may put things in perspective if you show how the battery consumed due to this stacks up with everything else.
Most of those who use xDrip like myself use it on their main phone they use for email, text, ...

I use xDrip 24/7 as a collector.
Looking at the battery usage, I can see this:
![Screenshot_20230121-080507](https://user-images.githubusercontent.com/51497406/213868419-4d8f8c87-c137-4e3d-8a9e-f0b422d3b721.png)
According to that, xDrip is draining my battery like there is no tomorrow.

But, if I add all the system apps to the list, this is what I get:
![Screenshot_20230121-080547](https://user-images.githubusercontent.com/51497406/213868451-d0f4377c-794a-4c6d-b4a5-36fcf02fceb5.png)

Now, it can be seen that in the grand scheme of things, the xDrip consumption is not really a concern.

I admit that not everyone uses their phone the same way.  But, I think it is a reasonable assumption that most people don't have a phone stacked in their purse or pocket just for xDrip while having a second phone on them for everything else.

We have limited resources.  We need to prioritize.  Unless I am misunderstanding something here, this does not seem like a priority.

Would you please explain why you think this needs to be addressed?

**Reply by nekromantik** [reply]
I didn't have theses on my older phone when I was using dexcom g6 so thought maybe g7 and companion mode was using more then g6 without companion mode

**Reply by Navid200** [reply]
That may be the case.
Even then, the only time there would be a reason for us to make a change to xDrip would be if the user would see a practical difference.
If your phone needs to be charged once a day now, and we change this as you desire and your phone still needs to be charged once a day, what would be the point?  How would that help you?

**Reply by jamorham** [reply]
The name of the wakelock is just where it was implemented originally for a code path that inserts data received from G5. It is used more generically now for various data inserts where we receive the value already calculated and it isn't a raw sensor reading. There is a lingering wakelock for each reading which is designed to ensure all subsystems get time to fire up, for example if there is web upload, sync to a watch, xDrip follower etc. You shouldn't see more than 4 seconds per reading which shouldn't equate to very much power usage over a 24 hour period.
